src/env.h

```diff
@@ -584,7 +584,7 @@ class IsolateData : public MemoryRetainer {
   inline v8::Local<v8::String> async_wrap_provider(int index) const;
 
   size_t max_young_gen_size = 1;
-  std::unordered_map<const char*, v8::Eternal<v8::String>> static_str_map;
+  std::unordered_map<const void*, v8::Eternal<v8::String>> static_str_map;
```

Comment:
This *is* a case in which the keys are strings (static strings, but that’s what the name is supposed to communicate) :)

Reply:
Right, but the key identity isn't based on string equality but on pointer identity (afaict at least). I assume if two static strings would be defined with the same "string value", they would create two separate entries in the map (because the map only cares about the pointer value).

I can definitely see how this is obvious behavior for somebody very familiar with C/C++ but it can be misleading for people coming from other languages that see "const char*" and think "string", leading to bad intuitions about `char*` map keys. Using `void*` loses type safety but may make it more obvious that the map only considers pointer equality.

I'll happily yield that this is in the realm of opinions. I found this because we internally floated a patch for this code because our C++ linter rules complained.